File: 0136-single-number.py
"""
136. Single Number

Runtime: 4 ms (beats 48.81%)
Memory: 19.51 MB (beats 6.17%)
"""

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        return functools.reduce(lambda x, y: x ^ y, nums)

File: 0136-single-number.cpp
/*
136. Single Number

Submitted: December 20, 2024

Runtime: 0 ms (beats 100.00%)
Memory: 20.71 MB (beats 23.16%)
*/

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int n = 0;
        for (const int i : nums) {
            n ^= i;
        }
        return n;
    }
};